app/models/mapping.rb

```diff
@@ -221,9 +237,55 @@ def dataset_path_must_exist
     errors.add(:base, "Dataset must exist.") unless dataset.exist?
   end
 
+  # Validator for app
+  def app_path_may_not_be_blank
+    errors.add(:base, "You must select an app.") unless app.exist?
+  end
+
   # Directories between $HOME and the configured database directory must be set to 775
   # @return [Boolean]
   def self.directory_perms_are_775?(directory)
     directory.stat.mode.to_s(8).end_with?('775')
   end
+
+  # Checks to see if FACLs are modifiable by the user
+  #
+  # Note that this check assumes that no negative permissions have been set.
+  # Also we are not using an ownership check since that could mask problems
+  # with permissions that would impact other users.
+  #
+  # @return [Boolean]
+  def self.can_modify_facl?(pathname)
+    pathname.exist? && group_facl_entry_has_C_set?(pathname)
+  end
+
+  # Get the group owner's name
+  # @return [String]
+  def self.get_groupname_for_pathname(pathname)
+    Etc.getgrgid(pathname.stat.gid).name
+  end
+
+  # Check if pathname is owned by the admin_group
+  # @return [Boolean]
+  def self.app_has_correct_group_ownership?(pathname)
+    get_groupname_for_pathname(pathname) == Configuration.admin_group
+  end
+
+  # Does the app have permission modification enabled for the GROUP principle
+  # @return [Boolean]
+  def self.group_facl_entry_has_C_set?(pathname)
+    result = false
+    begin
+      result = OodSupport::ACLs::Nfs4ACL.get_facl(
+        path: pathname
+      ).entries.select{
+        |entry| entry.principle == 'GROUP'
+      }.first.permissions.include?(:C)
+    rescue OodSupport::InvalidPath, OodSupport::BadExitCode
+    end
```

Comment:
Might be worth logging when this occurs, even if we are not raising an error. If we are debugging and looking through logs this could be useful to know. Do you know what situations `OodSupport::InvalidPath, OodSupport::BadExitCode` would occur?